### DISCUSSION TITLE
Clear all bot commands

### DISCUSSION THREAD

**Discussion by caiocinel** [Rapptz/discord.py#9064] [Q&A] [answered]
Recently I've been testing several different libraries to use the Discord API, so some commands were generated on these other platforms, but now I need to remove the commands registered by them, since they don't exist in the version using discord.py. I discovered the tree.clear_commands command, but it only excludes the commands that discord.py itself recorded. Is there any way to do this with all BOT commands?

Example commands (created using discord.js)
![image](https://user-images.githubusercontent.com/60858720/201441479-cd726dd3-25fa-4ceb-a447-6000714decea.png)

**Comment by Rapptz** [comment]
Call `sync()` on the tree.

**Comment by caiocinel** [comment]
```
from classes.Client import client
import discord

@client.tree.command()
async def settings(ctx: discord.Interaction):
    await ctx.response.send_message('Settings run')
    client.tree.clear_commands(guild=ctx.guild)
    await client.tree.sync(guild=ctx.guild)
```

After executing the code above, it only deletes the commands that discord.py created itself, the rest remain on the server

Before command:
![image](https://user-images.githubusercontent.com/60858720/201480279-7e07e87d-9a9c-43d3-a665-e089e6b47d78.png)

After:
![image](https://user-images.githubusercontent.com/60858720/201480299-3348ee94-ed2f-4918-80fb-29d75c21a58d.png)

**Comment by LeoCx1000** [comment, marked as answer]
they are probably global commands, sync but with ``guild=None``

**Reply by caiocinel** [reply]
That really was the problem, thanks!

**Reply by Midicix** [reply]
How do you use it ?

**Comment by jakobdylanc** [comment]
I was able to clear all bot commands by simply running these 2 lines (replace `discord_client` with your discord.Client instance):

```python
tree = discord.app_commands.CommandTree(discord_client)
await tree.sync()
```

**Comment by melzyy** [comment]
For those who just want a standalone solution to delete all commands for their bot with my extremely limited knowledge I've made this. Surprisingly, it even works.

```python
import discord

intents = discord.Intents.default()
client = discord.Client(intents=intents)
tree = discord.app_commands.CommandTree(client)

@client.event
async def on_ready():
    print(f'We have logged in as {client.user}')
    guilds = [guild.id for guild in client.guilds]
    print(f'The {client.user.name} bot is in {len(guilds)} Guilds.\nThe guilds IDs list: {guilds}')
    for guildId in guilds:
        guild = discord.Object(id=guildId)
        print(f'Deleting commands from {guildId}.....')
        tree.clear_commands(guild=guild,type=None)
        await tree.sync(guild=guild)
        print(f'Deleted commands from {guildId}!')
        continue
    print('Deleting global commands.....')
    tree.clear_commands(guild=None,type=None)
    await tree.sync(guild=None)
    print('Deleted global commands!')

client.run('BOT_TOKEN')
```

More experienced users are more than welcomed to improve this in any way you prefer.

**Reply by eth4n-dev** [reply]
I LOVE YOU BRO. This is the solution that worked for me. Reading through it I understand what the code does, but since I had made the bot in discordjs 5 years ago and have now been on discord.py for 30 minutes, I would've never figured it out myself. THANK YOU!
